packages/core/src/dialects/postgres/query-generator.js

```diff
@@ -167,8 +167,8 @@ export class PostgresQueryGenerator extends PostgresQueryGeneratorTypeScript {
     return `DROP TABLE IF EXISTS ${this.quoteTable(tableName)}${options?.cascade ? ' CASCADE' : ''};`;
   }
 
-  showTablesQuery() {
-    const schema = this.options.schema || 'public';
+  showTablesQuery(showTableSchema) {
+    const schema = showTableSchema || this.options.schema || 'public';
 
```

Comment:
```suggestion
  showTablesQuery(schema) {
```

You are passing the correct schema in `QueryInterface.showAllTables` already

Reply:
Yup, removed this. 

'public' is defined as the default schema in `getDefaultSchema` for postgres.